internal/security/secretstore/tokenfilewriter/tokenfilewriter.go

```diff
@@ -0,0 +1,172 @@
+		}
+	}()
+
+	// Write the token issuing token to disk to pass it to the token provider
+	tokenFileAbsPath, fileErr := filepath.Abs(tokenFilePath)
+	if fileErr != nil {
+		return nil, fmt.Errorf("failed to convert tokenFile to absolute path %s: %s", tokenFilePath, fileErr.Error())
+	}
+
+	dirOfCreatedToken := filepath.Dir(tokenFileAbsPath)
+	fileErr = w.fileOpener.MkdirAll(dirOfCreatedToken, 0700)
+	if fileErr != nil {
+		return nil, fmt.Errorf("failed to create tokenpath base dir: %s", fileErr.Error())
+	}
+
+	fileWriter, fileErr := w.fileOpener.OpenFileWriter(tokenFileAbsPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+	if fileErr != nil {
+		return nil, fmt.Errorf("failed to create token file %s: %s", tokenFileAbsPath, fileErr.Error())
+	}
+
+	if fileErr = json.NewEncoder(fileWriter).Encode(tokenCreated); fileErr != nil {
+		_ = fileWriter.Close()
+		return nil, fmt.Errorf("failed to write created token: %s", fileErr.Error())
+	}
+
+	if fileErr = fileWriter.Close(); fileErr != nil {
+		return nil, fmt.Errorf("failed to close token file: %s", fileErr.Error())
+	}
+
+	w.logClient.Infof("token's written to %s", tokenFilePath)
+
+	return revokeTokenFunc, nil
+}
+
+// CreateMgmtTokenForConsulSecretsEngine creates a new Vault token that
+// allows the Consul bootstrapper to operate on managing Vault's Consul secrets engine related APIs (see reference:
+// https://www.vaultproject.io/api-docs/secret/consul). The created Vault token is meant for serving
+// the purpose of Consul ACL's bootstrapping as part of securing Consul process.
+//
+// Requires a root token to create, and returns data/information containing the token,
+// keeping the token without revoking it and hence always returning nil RevokeFunc;
+// also returns non-nil error if anything goes wrong during the creation.
+// this function conforms to the signature of the tokencreatable.CreateTokenFunc type
+// so that it can be passed to CreateAndWrite()
+func (w TokenFileWriter) CreateMgmtTokenForConsulSecretsEngine(rootToken string) (map[string]interface{},
+	tokencreatable.RevokeFunc, error) {
+	consulSecretsEngineOpsPolicyDocument := `
+# allow to configure the access information for Consul
+path "` + secretsengine.ConsulSecretEngineMountPoint + `/config/access" {
+    capabilities = ["create", "update"]
+}
+
+# allow to create, update, read, list, or delete the Consul role definition
+path "` + secretsengine.ConsulSecretEngineMountPoint + `/roles/*" {
+    capabilities = ["create", "read", "update", "delete", "list"]
+}
+
+`
+
+	if err := w.secretClient.InstallPolicy(rootToken,
+		consulSecretsEngineOpsPolicyName,
+		consulSecretsEngineOpsPolicyDocument); err != nil {
+		return nil, nil, fmt.Errorf("failed to install Consul secrets engine operations policy: %v", err)
+	}
+
+	// setup new token's properties
+	tokenParams := make(map[string]interface{})
+	tokenParams["type"] = "service"
+	tokenParams["display_name"] = "managing with policy name " + consulSecretsEngineOpsPolicyName
```

Comment:
Suggest display_name = "Configuration token for consul secrets engine"

Reply:
i think our go.mod.secret code in Vault's CreateToken() automatically prefaces `token-`  -xxxxxx something or Vault itself does it, so with my example above, the display_name will be `token-managing-with-policy-name-xxxxxxx`,  that is why I keep the display name this way.

Reply:
"token" + "display_name":  **for Consul ACL bootstrap**
resolved